[26.x] Changes for rc3

fanquake opened the pull request:
Currently backports:
* https://github.com/bitcoin/bitcoin/pull/28858
* https://github.com/bitcoin/bitcoin/pull/28895 (partial)
* https://github.com/bitcoin/bitcoin/pull/28913
* https://github.com/bitcoin/bitcoin/pull/28905
* https://github.com/bitcoin/bitcoin/pull/28919
* https://github.com/bitcoin/bitcoin/pull/28925


Also includes changes for rc3, and reintegrating the release-notes.

fanquake committed:
doc: rewrite explanation for -par=

The negative bound for script threads comes from the machine which
generates the man pages, so may only be correct for that machine. Any
other placeholder value will also be wrong for some machines. Fix this
be removing the value. This also fixes help2man incorrectly bolding the
value, as if it were a paramater.

Closes #28850.

Github-Pull: #28858
Rebased-From: d799ea26edfd63434b6d1cf55500de184dc67fac

fanquake committed:
doc: update release notes for v26.0rc3

Few further changes are expected, so reintegrate the release-notes.

fanquake committed:
build: bump version to v26.0rc3

fanquake committed:
doc: update manual pages for v26.0rc3

fanquake committed:
doc: regenerate example bitcoin.conf

jonatack committed:
p2p: do not make automatic outbound connections to addnode peers

to allocate our limited outbound slots correctly, and to ensure addnode
connections benefit from their intended protections.

Our addnode logic usually connects the addnode peers before the automatic
outbound logic does, but not always, as a connection race can occur.  If an
addnode peer disconnects us and if it was the only one from its network, there
can be a race between reconnecting to it with the addnode thread, and it being
picked as automatic network-specific outbound peer.  Or our internet connection
or router, or the addnode peer, could be temporarily offline, and then return
online during the automatic outbound thread.  Or we could add a new manual peer
using the addnode RPC at that time.

The race can be more apparent when our node doesn't know many peers, or with
networks like cjdns that currently have few bitcoin peers.

When an addnode peer is connected as an automatic outbound peer and is the only
connection we have to a network, it can be protected by our new outbound
eviction logic and persist in the "wrong role".

Examples on mainnet using logging added in the same pull request:

2023-08-12T14:51:05.681743Z [opencon] [net.cpp:1949] [ThreadOpenConnections]
[net:debug] Not making automatic network-specific outbound-full-relay connection
to i2p peer selected for manual (addnode) connection: [geh...odq.b32.i2p]:0

2023-08-13T03:59:28.050853Z [opencon] [net.cpp:1949] [ThreadOpenConnections]
[net:debug] Not making automatic block-relay-only connection to onion peer
selected for manual (addnode) connection: kpg...aid.onion:8333

2023-08-13T16:21:26.979052Z [opencon] [net.cpp:1949] [ThreadOpenConnections]
[net:debug] Not making automatic network-specific outbound-full-relay connection
to cjdns peer selected for manual (addnode) connection: [fcc...8ce]:8333

2023-08-14T20:43:53.401271Z [opencon] [net.cpp:1949] [ThreadOpenConnections]
[net:debug] Not making automatic network-specific outbound-full-relay connection
to cjdns peer selected for manual (addnode) connection: [fc7...59e]:8333

2023-08-15T00:10:01.894147Z [opencon] [net.cpp:1949] [ThreadOpenConnections]
[net:debug] Not making automatic feeler connection to i2p peer selected for
manual (addnode) connection: geh...odq.b32.i2p:8333

Finally, there does not seem to be a reason to make block-relay or short-lived
feeler connections to addnode peers, as the addnode logic will ensure we connect
to them if they are up, within the addnode connection limit.

Fix these issues by checking if the address is an addnode peer in our automatic
outbound connection logic.

Github-Pull: #28895
Rebased-From: cc627169206fe902157806d88fcaf2b05701c38d

hebasto committed:
ci: Avoid toolset ambiguity that MSVC can't handle

This change is required to work with the new windows-2022 image version
20231115 properly.

Github-Pull: #28905
Rebased-From: 91d5bd8ac9a28725c735f8e6900bc85673bb190a

martinus committed:
pool: make sure PoolAllocator uses the correct alignment

This changes the PoolAllocator to default the alignment to the given type. This makes the code simpler, and most importantly
fixes a bug on ARM 32bit that caused OOM: The class CTxOut has a member CAmount which is an int64_t and on ARM 32bit int64_t
are 8 byte aligned which is larger than the pointer alignment of 4 bytes. So for CCoinsMap to be able to use the pool, we
need to use the alignment of the member instead of just alignof(void*).

Github-Pull: #28913
Rebased-From: ce881bf9fcb7c30bb1fafd6ce38844f4f829452a

martinus committed:
pool: change memusage_test to use int64_t, add allocation check

If alignment of the PoolAllocator would be insufficient, then the test would fail. This also catches the issue with ARM 32bit,
where int64_t is aligned to 8 bytes but void* is aligned to 4 bytes. The test adds a check to ensure the pool has allocated
a minimum number of chunks

Github-Pull: #28913
Rebased-From: d5b4c0b69e543de51bb37d602d488ee0949ba185

hebasto commented on doc/release-notes.md:
```diff
@@ -1 +1,327 @@
+
+- It's now possible to use [Miniscript](https://bitcoin.sipa.be/miniscript/) inside Taproot leaves for descriptor wallets. (#27255)
+
+GUI changes
+-----------
```
I'm sorry for that late comment. Mind adding the following stuff:
```
- The transaction list in the GUI to no longer provide a special category for “payment to yourself”. Now transactions that have both inputs and outputs that affect the wallet are displayed on separate lines for spending and receiving. (gui#119)

- A new menu option allows migrating a legacy wallet based on keys and implied output script types stored in BerkeleyDB (BDB) to a modern wallet that uses descriptors stored in SQLite. (gui#738)

- The PSBT operations dialog marks outputs paying your own wallet with “own address”. (gui#740)

- The ability to create legacy wallets is being removed. (gui#764)
```
?

fanquake replied:
Sure. Added to the next push.

hebasto committed:
build: Fix regression in "ARMv8 CRC32 intrinsics" test

The `vmull_p64` is a part of the Crypto extensions from the ACLE. They
are optional extensions, so they get enabled with a `+crypto` for
architecture flags.

Github-Pull: #28919
Rebased-From: 228d6a2969e4fcee573c9df7aad31550eab9c8d4

hebasto committed:
ci: Update apt cache

Github-Pull: #28925
Rebased-From: a6cc059ea50b3629a287f66be3658e766b6ddd47

hebasto committed:
ci: Switch from `apt` to `apt-get`

Github-Pull: #28925
Rebased-From: 710da28c72259e1054bb63d2d81a8f3aea4c9000

hebasto commented:
Suggesting to backport https://github.com/bitcoin/bitcoin/pull/28905 as well to ensure CI pass.

hebasto commented on doc/release-notes.md:
```diff
@@ -1 +1,335 @@
+GUI changes
+-----------
+
+- The transaction list in the GUI to no longer provide a special category for "payment to yourself". Now transactions that have both inputs and outputs that affect the wallet are displayed on separate lines for spending and receiving. (gui#119)
```
nit: 
```suggestion
- The transaction list in the GUI no longer provides a special category for "payment to yourself". Now transactions that have both inputs and outputs that affect the wallet are displayed on separate lines for spending and receiving. (gui#119)
```

I apologize for the wrong Grammar in my initial suggestion.

fanquake replied:
No worries, fixed up.

hebasto reviewed (approved):
ACK 740a85ce5f25f2282e32f148d61f92301bef7179, I've verified backports locally, also reviewed changes related to the `-par` option and `rc2` --> `rc3` as well.

hebasto reviewed (approved):
re-ACK f868852d3d7321aa522d6b168f33bef1e5eba2dc.

hebasto reviewed (approved):
re-ACK 2f86d3053314c382dfce5cf314e98811ed433859, only https://github.com/bitcoin/bitcoin/pull/28919 backported since my [recent](https://github.com/bitcoin/bitcoin/pull/28872#pullrequestreview-1744950215) review.

sedited reviewed (approved):
ACK 2f86d3053314c382dfce5cf314e98811ed433859

Mita23456 reviewed (commented):
So not doing at

Mita23456 commented on src/bench/pool.cpp:
```diff
@@ -37,8 +37,7 @@ static void PoolAllocator_StdUnorderedMapWithPoolResource(benchmark::Bench& benc
                                    std::hash<uint64_t>,
```
Ok

Mita23456 commented on src/bench/pool.cpp:
```diff
@@ -37,8 +37,7 @@ static void PoolAllocator_StdUnorderedMapWithPoolResource(benchmark::Bench& benc
+                                                 sizeof(std::pair<const uint64_t, uint64_t>) + 4 * sizeof(void*)>>;
 
     // make sure the resource supports large enough pools to hold the node. We do this by adding the size of a few pointers to it.
     auto pool_resource = Map::allocator_type::ResourceType();
```
Yep

Mita23456 reviewed (commented):
Hello
